src/sagemaker/jumpstart/cache.py

```diff
@@ -129,15 +146,44 @@ def get_region(self) -> str:
         """Return region for cache."""
         return self._region
 
-    def set_manifest_file_s3_key(self, key: str) -> None:
-        """Set manifest file s3 key. Clears cache after new key is set."""
-        if key != self._manifest_file_s3_key:
-            self._manifest_file_s3_key = key
-            self.clear()
+    def set_manifest_file_s3_key(
+        self,
+        key: str,
+        file_type: JumpStartS3FileType = JumpStartS3FileType.MANIFEST,
+    ) -> None:
+        """Set manifest file s3 key. Clears cache after new key is set.
 
-    def get_manifest_file_s3_key(self) -> str:
+        Raises:
+            ValueError: if hte file type is not recognized
+        """
```

Comment:
along the lines of the comment on set_manifest_file_s3_key, this logic can be simplified